### PR TITLE
Make subcommand for semantic clusters algorithm

### PR DESCRIPTION
**Pull request by jteijema**
This PR will turn the semantic clustering into an ASReview entrypoint. It should also remove hardcoded file storage. Finally, as from this point on it will be an asreview plugin, it will implement github workflow testing!

### CODE REVIEW

**Comment by jteijema** [comment]
Merge after #22 and #23 are merged! This PR creates the ASReview entrypoint.

image

**Comment by jteijema** [comment]
Final PR will be the file storage implementation

**Comment by J535D165** [comment]
> Merge after #22 and #23 are merged! This PR creates the ASReview entrypoint.
> 
> image

This command doesn't return the help, but the error ;) Consider renaming the entry point.

**Comment by jteijema** [comment]
Renaming will be done all at once in a future PR. I want to rename everything!

**Comment by jteijema** [comment]
Ready to merge!

File: asreviewcontrib/semantic_clustering/main.py [hunk 1]
```diff
@@ -2,51 +2,79 @@
...
-            filepath = arg
-        elif opt in ("-t", "--testfile"):
-            filepath = "https://raw.githubusercontent.com/asreview/systematic-review-datasets/master/datasets/van_de_Schoot_2017/output/van_de_Schoot_2017.csv"
-        elif opt in ("-a", "--app"):
+from asreviewcontrib.semantic_clustering.semantic_clustering import SemanticClustering  # noqa: E501
+
+
+class SemClusEntryPoint(BaseEntryPoint):
+    description = "Semantic clustering tools for ASReview."
+    extension_name = "asreview-semantic-clustering"
```
Comment (J535D165) [inline asreviewcontrib/semantic_clustering/main.py]:
```suggestion
    extension_name = "semantic_clustering"
```

File: asreviewcontrib/semantic_clustering/main.py [hunk 2]
```diff
@@ -2,51 +2,79 @@
...
-    # check if arguments are empty
-    if filepath == "":
-        print('Please use the following format:')
-        print('test.py -f <filepath>')
-        print('test.py --testfile')
-        print('test.py --app')
-        sys.exit(2)
+# argument parser
+def _parse_arguments(version="Unknown", argv=None):
+    parser = argparse.ArgumentParser(prog='asreview semantic clustering')
```
Comment (J535D165) [inline asreviewcontrib/semantic_clustering/main.py]:
```suggestion
    parser = argparse.ArgumentParser(prog='asreview semantic_clustering')
```

File: setup.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        "seaborn",
+        "torch",
+    ],
+
+    extras_require={
+    },
+
+    entry_points={
+        "asreview.entry_points": [
+            "semantic-clustering = asreviewcontrib.semantic_clustering.main:SemClusEntryPoint",  # noqa: E501
```
Comment (J535D165) [inline setup.py]:
```suggestion
            "semantic_clustering = asreviewcontrib.semantic_clustering.main:SemClusEntryPoint",  # noqa: E501
```

**Review by J535D165 — CHANGES_REQUESTED** [review]
I don't feel like merging bugs/broken code into main is a good idea. I'm suggesting changes (and rename afterward).

**Review by J535D165 — APPROVED** [review]

**Comment by J535D165** [comment]
Thanks!

**Pull request merged**